README.md

```diff
@@ -1,96 +1,354 @@
+```
+
+### Custom verification
+
+By default the provided `verify` method performs multiple checks on a document
+
+- for the type `DOCUMENT_STATUS`: it runs `OpenAttestationEthereumDocumentStoreStatus`, `OpenAttestationEthereumTokenRegistryStatus` and `DidSignedDocumentStatus` verifiers
+- for the type `DOCUMENT_INTEGRITY`: it runs `OpenAttestationHash` verifier
+- for the type `ISSUER_IDENTITY`: it runs `OpenAttestationDnsTxt`, `DnsDidProof` and `DidIdentityProof` verifiers
+
+All those verifiers are exported as `openAttestationVerifiers`
+
+You can build your own verify method or your own verifiers:
+
+```ts
+// creating your own verify using default exported verifiers
+import { verificationBuilder, openAttestationVerifiers } from "@govtechsg/oa-verify";
+
+const verify1 = verificationBuilder(openAttestationVerifiers, { network: "ropsten" }); // this verify is equivalent to the one exported by the library
+// this verify is equivalent to the one exported by the library
+const verify2 = verificationBuilder([openAttestationVerifiers[0], openAttestationVerifiers[1]], {
+  network: "ropsten",
+}); // this verify only run 2 verifiers
+```
+
+```ts
+// creating your own verify using custom verifier
+import { verificationBuilder, openAttestationVerifiers, Verifier } from "@govtechsg/oa-verify";
+const customVerifier: Verifier = {
+  skip: () => {
+    // return a SkippedVerificationFragment if the verifier should be skipped or throw an error if it should always run
   },
-  {
-    "name": "OpenAttestationDidSignedDocumentStatus",
-    "reason": {
-      "code": 0,
-      "codeString": "SKIPPED",
-      "message": "Document was not signed by DID directly"
-    },
-    "status": "SKIPPED",
-    "type": "DOCUMENT_STATUS"
+  test: () => {
+    // return true or false
   },
-  {
-    "data": [
-      {
-        "location": "example.tradetrust.io",
-        "status": "VALID",
-        "value": "0xe59877ac86c0310e9ddaeb627f42fdee5f793fbe"
-      }
-    ],
-    "name": "OpenAttestationDnsTxtIdentityProof",
-    "status": "VALID",
-    "type": "ISSUER_IDENTITY"
+  verify: async (document) => {
+    // perform checks and returns a fragment
   },
-  {
-    "name": "OpenAttestationDnsDidIdentityProof",
-    "reason": {
-      "code": 0,
-      "codeString": "SKIPPED",
-      "message": "Document was not issued using DNS-DID"
-    },
-    "status": "SKIPPED",
-    "type": "ISSUER_IDENTITY"
+};
+
+// create your own verify function with all verifiers and your custom one
+const verify = verificationBuilder([...openAttestationVerifiers, customVerifier], { network: "ropsten" });
+```
+
+Refer to [Extending Custom Verification](#extending-custom-verification) to find out more on how to create your own custom verifier.
+
+### Custom validation
+
+The `isValid` function will execute over fragments and determine if the fragments produced a valid result. By default the function will return true if a document fulfill the following conditions:
```

Comment:
Maybe we should add a sentence first to remind that verification will return fragments (and what is a fragment_

Reply:
In a recent adr, I added definitions. Could be useful here too: https://github.com/Open-Attestation/adr/blob/master/decentralised_rendering.md#definitions